p2p/switch.go

```diff
@@ -598,8 +601,9 @@ func (sw *Switch) AddUnconditionalPeerIDs(ids []string) error {
 	for i, id := range ids {
 		err := validateID(ID(id))
 		if err != nil {
-			return fmt.Errorf("wrong ID #%d: %w", i, err)
+			return ErrInvalidPeerID{ID: ID(id), source: fmt.Errorf("wrong ID #%d: %w", i, err)}
```

Comment:
I assume the `err` should be passed here and formatted at the error type.

Reply:
Updated.